### ISSUE TITLE
karmabot

### ISSUE THREAD

**Issue by aromatt** [aromatt/thejub.pub#177]
some reference code courtesy of @atrivedi https://github.com/atrivedi/slack-slash/blob/master/karma.js

**Comment by atrivedi** [comment]
I can tackle this. it'll be super simple since i know what the code does. Hopefully in the next few days

**Comment by aromatt** [comment]
I think it would be cool to see it broken down by user and type of karma, like this:
```
aromatt's karma DB entry:
{
  'chat': {
    'jeskaeli': 5, # jeskaeli has given me 5 chat karma
    'ankit': 1000, # ankit has given me 1000...
  }
  'video-plays': {
    'ankit': 1,
    'BaronPumpky': 3,
  }
}
```

**Comment by atrivedi** [comment]
I don't understand that

**Comment by aromatt** [comment]
It would be cool to be able to see who likes who the most

**Comment by aromatt** [comment]
but the bot would still only report the sum.  Seeing it broken down like that in th UI would come later or never.

**Comment by aromatt** [comment]
the schema I posted is a proposal for what to store in mongodb

**Comment by aromatt** [comment]
more thinking out loud:
I have been wanting to track video-hoots with fine granularity, like one mongo document per hoot:
```
{
  'dj': 'aromatt',
  'hooted-by': 'ankit',
  'video-id': 'jfl38f_83jhkd'
}
```
which that makes this conversation a little hairier.

**Comment by aromatt** [comment]
after more discussion I think we should have one karma collection in mongodb, and store every single karma event as a new document (this would be the most mongo-y thing to do IMO).  Then, when the bot needs to report "aromatt has X karma", we'd just do a quick DB aggregation (just count all the karma documents which have "aromatt" as the recipient).  Storing them this way would make it so there's only one source of truth, and would let us do more interesting analysis later, like "aromatt likes jeskaeli's videos the most" or "aromatt likes acowinthecrowd's chat messages the most"

**Comment by aromatt** [comment]
```
{
  'type': 'video',
  'value': 1,
  'recipient': 'aromatt',
  'giver': 'ankit',
  'context': 'oKDU+/kd9',
  'timestamp': <Date>
}
```

**Comment by aromatt** [comment]
implemented by edc00ba7f26d31457205cbab1c7ccb7899a8afee
